Pidaq/Lib/wind_tunnel/air_properties.py

```diff
@@ -0,0 +1,112 @@
+"""Contains class that defines linear interpolation functions for wind tunnel testing
+
+Classes:
+    LinearInterpolation
+"""
+
+import csv
+import logging
+import sys
+from scipy import interpolate
+
+class LinearInterpolation:
+    """Creates linear interpolation functions for density and dynamic viscosity
+
+    This class defines two functions that are used to linearly interpolate
+    density (kg/m3) and dynamic viscosity (Pa*S) based on temperature (°C)
+    for the purpose of wind tunnel testing.
+    
+    Attributes:
+        densityFunc
+        viscosityFunc
+
+    Methods:
+        interpolateDensity
+        interpolateViscosity
+    """
+
+    def __init__(self, path):
+        """
+        Reads a CSV file to define two functions for density and dynamic
+        viscosity that can be used with temperature.  The CSV should have same
+        structure as current one in Drive, where the units are in the second row.
+        
+        Parameters:
+            path - path to a CSV file of the air properties
+        """
+        self.logger = logging.getLogger()
+
+        try:
+            open(path) # Open the CSV file
+        except IOError as err:
+            self.logger.error(err)
+            sys.exit()
+        with open(path) as file:
+            csv_reader= csv.reader(file, delimiter =',')
+            next(csv_reader) # Skip the first row
+            tempCol, densityCol, viscosityCol = -1,-1,-1
+            row = next(csv_reader) # Read the second row that contains the units
+        
+            count = 0 # For tracking the column index
+            for cell in row: # This loop determines the column index for 
+                    # temperature, density, and dynamic viscosity based on units
+
+                if cell.find('°C') != -1:
+                    tempCol = count
+
+                if cell.find('[kg/m3]') != -1:
+                    densityCol = count
+
+                if cell.find('[Pa*S]') != -1:
+                    viscosityCol = count
+                count += 1
```

Comment:
Just a cool thing you can do here instead of creating and incrementing the counter manually, you can do an enumeration that keeps a running tally of the loop index (what you have there is fine tho)

Its something like:

``` For index, cell in enumerate(row): ```

Then you can simply use `tempCol = index` and so on. 

Reference:
[https://www.programiz.com/python-programming/methods/built-in/enumerate](https://www.programiz.com/python-programming/methods/built-in/enumerate)

Reply:
I did know about enumerate before. I think it is brilliant. I decided to give it a go and seems like it is working.